### PR TITLE
feat: [CDS-86868] search and pagination for infra definitions

### PR DESCRIPTION
**Pull request by kanikamittal09**
### Summary

- Added pagination support https://harness.atlassian.net/browse/CDS-86868
- Added search support https://harness.atlassian.net/browse/CDS-86867

#### Screenshots

- Pagination

https://github.com/harness/harness-core-ui/assets/94995410/a32a5338-a0cb-4cf9-8964-99a5d2d4fe6b

- Search 

https://github.com/harness/harness-core-ui/assets/94995410/62c70ed0-4d3c-4d5d-a3ef-a4818c404266

### Manual Testing
- created new env, new infra check for infra definitions listing 
- check for appropriate search result based on search term
- checked for page size and page navigation

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by kanikamittal09** [comment]
trigger messagecheck

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx [hunk 1]
```diff
@@ -136,32 +168,46 @@ export default function InfrastructureDefinition({ isEnvPage }: { isEnvPage: boo
...
+                  permission: PermissionIdentifier.EDIT_ENVIRONMENT
+                }}
+              />
+              <ExpandingSearchInput
+                alwaysExpanded
+                width={200}
+                placeholder={getString('search')}
+                onChange={
+                  /* istanbul ignore next */ text => {
+                    setSearchTerm(text.trim())
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx:198]: Can text be null or undefined here? In that case we should add null check

Comment (ishantmehtaharness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx:198]: else `.trim()` will fail

**Review by ishantmehtaharness — DISMISSED** [review]
looks good, with `.trim()` check, please check that

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx [hunk 2]
```diff
@@ -8,31 +8,45 @@
...
+import { DEFAULT_PAGE_INDEX } from '@pipeline/utils/constants'
+import { useRbacQueryParamOptions } from '@rbac/utils/utils'
 import InfrastructureList from './InfrastructureList/InfrastructureList'
 import InfrastructureModal from './InfrastructureModal'
 
 import css from './InfrastructureDefinition.module.scss'
 
 // TODO: To be removed once pagination and search support is available for Infrastructure Definitions
 // Ticket - https://harness.atlassian.net/browse/CDS-86868
-const DEFAULT_PAGE_SIZE_FOR_INFRASTRUCTURES = 1000
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx:35]: Do not make these changes. Keep this. This has been done to load more templates.

**Review by pankajsingh-harness — CHANGES_REQUESTED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx (hunk 2 above)
Comment (kanikamittal09) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx:35]: discussed offline , changes only related to infra listing page

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx (hunk 1 above)
Comment (kanikamittal09) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/InfrastructureDefinition.tsx]: changed it to `text?.trim()`

**Review by ishantmehtaharness — APPROVED** [review]
Looks good

**Review by pankajsingh-harness — APPROVED** [review]

**Pull request merged**